include/mamba/config.hpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2019, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#ifndef MAMBA_CONFIG_HPP
+#define MAMBA_CONFIG_HPP
+
+#include "mamba/context.hpp"
+#include "mamba/mamba_fs.hpp"
+#include "mamba/output.hpp"
+#include "mamba/util.hpp"
+
+#include <yaml-cpp/yaml.h>
+
+
+namespace mamba
+{
+    namespace detail
+    {
+        void load_rc(const YAML::Node& config)
+        {
+            auto& ctx = Context::instance();
+            auto new_channels = config["channels"].as<std::vector<std::string>>();
+
+            for (auto& channel : new_channels)
+            {   // Remove duplicates
+                auto c = std::find(ctx.channels.begin(), ctx.channels.end(), channel);
+                if (c != ctx.channels.end())
+                {
+                    ctx.channels.erase(c);
+                }
+            }
+
+            ctx.channels.insert(ctx.channels.begin(), new_channels.begin(), new_channels.end());
```

Comment:
I think it would be better to have this function somewhere else, because we need to merge channels from the command line as well.

Also FYI there is the `--override-channels` option on the `conda` command line that we'll have to handle as well.

Looks good though!

Reply:
Also I just realized that there is an additional complication because the same channel can appear in "different" forms:

- conda-forge
- https://conda.anaconda.org/conda-forge

is the same channel. So that function should probably normalize the channel names to URLs and then make sure that they are unique and sorted in the right way.

Anyways, we can keep that also for later.

Reply:
FYI we have the code for this normalization in channel.hpp

Reply:
Thanks a lot for your comments @wolfv !
I will improve this first shot to handle those cases